servicetalk-opentelemetry-http/src/main/java/io/servicetalk/opentelemetry/http/OpenTelemetryOptions.java

```diff
@@ -25,25 +28,53 @@
 
 import java.util.ArrayList;
 import java.util.List;
+import java.util.function.Function;
+import javax.annotation.Nullable;
 
 import static java.util.Collections.emptyList;
 import static java.util.Collections.unmodifiableList;
+import static java.util.Objects.requireNonNull;
 
-/**
- * A set of options for configuring OpenTelemetry filters.
- */
+/** A set of options for configuring OpenTelemetry filters. */
 public final class OpenTelemetryOptions {
 
+    @Nullable
+    private final Function<HttpRequestMetaData, String> spanNamePrefix;
     private final List<String> capturedRequestHeaders;
     private final List<String> capturedResponseHeaders;
     private final boolean enableMetrics;
+    private final OpenTelemetry openTelemetry;
+    private final boolean ignoreSpanSuppression;
+
+    // Client-specific options (ignored by server filters)
+    private final String componentName;
 
-    OpenTelemetryOptions(final List<String> capturedRequestHeaders,
-                         final List<String> capturedResponseHeaders,
-                         final boolean enableMetrics) {
+    OpenTelemetryOptions(
+            @Nullable
+            final Function<HttpRequestMetaData, String> spanNamePrefix,
+            final List<String> capturedRequestHeaders,
+            final List<String> capturedResponseHeaders,
+            final boolean enableMetrics,
+            final OpenTelemetry openTelemetry,
+            final String componentName,
+            final boolean ignoreSpanSuppression) {
+        this.spanNamePrefix = spanNamePrefix;
         this.capturedRequestHeaders = capturedRequestHeaders;
         this.capturedResponseHeaders = capturedResponseHeaders;
         this.enableMetrics = enableMetrics;
+        this.openTelemetry = openTelemetry;
+        this.componentName = componentName;
+        this.ignoreSpanSuppression = ignoreSpanSuppression;
+    }
+
+    /**
+     * The span name prefix generator.
+     *
+     * @return a function that will extract a span name prefix from the {@link HttpRequestMetaData}.
+     */
+    @Nullable
+    Function<HttpRequestMetaData, String> spanNamePrefix() {
+        return spanNamePrefix;
     }
```

Comment:
I suspect this is probably better as a full span-name-extractor, and not just a prefix.